multipaz-models/src/commonMain/kotlin/org/multipaz/models/presentment/PresentmentModel.kt

```diff
@@ -36,6 +37,15 @@ class PresentmentModel {
         private const val TAG = "PresentmentModel"
     }
 
+    private var zkSystemRepository: ZkSystemRepository? = null
+
+    /**
+     * Sets the ZK System Repository to use for generating ZK proofs.
+     */
+    fun setZkSystemRepository(zkSystemRepository: ZkSystemRepository) {
+        this.zkSystemRepository = zkSystemRepository
+    }
+
```

Comment:
The `ZkSystemRepository` should come from the `PresentmentSource` the application connects later on so please move it there. This is already used to convey the `DocumentStore`, `DocumentTypeRepository`, and `TrustManager` used for reader authentication and it's only natural the application would provide the `ZkSystemRepository` to use, if any.

Should be a small change, suggest to just add `open val zkSystemRepository: ZkSystemRepository?` to `PresentmentSource` and ditto in `SimplePresentmentSource`. Then adjust `getPresentmentSource()` in App.kt to provide it.